xdsl/dialects/riscv.py

```diff
@@ -57,6 +59,11 @@ class Register:
     name: str | None = field(default=None)
     """The register name. Should be one of `ABI_INDEX_BY_NAME` or `None`"""
 
+    ABI_INDEX_BY_NAME: dict[str, int] = field(init=False)
```

Comment:
```suggestion
    abi_index_by_name: ClassVar[dict[str, int]] = field(init=False)
```

Reply:
it might be better off as an abstract class function than a class variable TBH, I don't think it's useful to be able to specify this value at runtime.